pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -56,8 +56,9 @@ type ManagerImpl struct {
 	socketname string
 	socketdir  string
 
-	endpoints map[string]endpoint // Key is ResourceName
-	mutex     sync.Mutex
+	endpointsRegistering map[string]endpointInfo // Key is ResourceName
```

Comment:
So that the plugin is notified of registration only before it is registered in the kubelet

Reply:
where in the code, endpoint stored in endpointsRegistering, is being retrieved?

Reply:
nevermind, its in registerPlugin.

Reply:
Why dont you merge init and register calls into one? anyways from watcher these apis are being called in succession right after each other. By keeping these seperate, unnecessarily, you need endpointsRegistering for holding the endpoint temporarily, otherwise it wont be required.

Reply:
Done